pkg/virt-handler/vm.go

```diff
@@ -2692,8 +2692,7 @@ func (d *VirtualMachineController) finalizeMigration(vmi *v1.VirtualMachineInsta
 }
 
 func isACPIEnabled(vmi *v1.VirtualMachineInstance, domain *api.Domain) bool {
-	zero := int64(0)
-	return vmi.Spec.TerminationGracePeriodSeconds != &zero &&
+	return (vmi.Spec.TerminationGracePeriodSeconds != nil && *vmi.Spec.TerminationGracePeriodSeconds != 0) &&
```

Comment:
Maybe this will fix it the test
```
if ((vmi.Spec.TerminationGracePeriodSeconds != nil && *vmi.Spec.TerminationGracePeriodSeconds != 0) || dom.Spec.Metadata.KubeVirt.GracePeriod.DeletionGracePeriodSeconds != 0) &&```

Reply:
i changed this just a little, to only check domain spec if vmi isn't defined. Thanks for all the help and insight!